tests/test_producer.py

```diff
@@ -113,14 +113,14 @@
         'test1',
         loop,
     )
-    mocker.spy(producer, '_complete_produce_request')
+    producer._complete_produce_request = MagicMock()
```

Comment:
This breaks the point of this test which is to make sure we call _complete_produce_request is called with the right number of arguments. `spy` will make sure the right number of arguments are passed in and execute the function, MagicMock will accept any and all arguments